plotly/tools.py

```diff
@@ -1443,6 +1443,376 @@ class FigureFactory(object):
     """
 
     @staticmethod
+    def create_scatterplotmatrix(df, useindex = False, index = " ",
+                                 diagonal = "Scatter", symbol = 0, size = 6,
+                                 height = " ", width = " ", jitter = 0,
+                                 title = "Scatterplot Matrix"):
+
+        from plotly import tools
+        import plotly.plotly as py
+        import plotly.graph_objs as go
+        import pandas as pd
+        import random as r
```

Comment:
In general, it's a good idea to import things at the top of a file, when possible, not dynamically like this.

AFAIK, the only thing that *needs* to be imported down here is https://github.com/plotly/plotly.py/blob/master/plotly/tools.py#L1743-L1744

Additionally, we protect `pandas` imports (among other things) so that we don't need to bring the big dependency into our `setup.py` file.

Can you first add a conditional check for [`_pandas_imported`](https://github.com/plotly/plotly.py/blob/master/plotly/tools.py#L54) and then raise a helpful warning instead of importing pandas here?

Finally, I think we typically just `import random`, without changing the namespace to `r`.

Some of this stuff is a little confusing, let me know if you have questions.